README.md

```diff
@@ -20,7 +20,7 @@ Package versions produced by official builds for this repo will automatically co
 The default build will automatically build all of the libraries that have been added to the repo. That said, given this repository is used to service packages, and you won't always want to service all of the packages at the same time, packages for libraries are not built by default. Whenever a package needs to be serviced, the following steps will need to be follwed:
 
 1. Determine the ServiceVersion to be used. When servicing a fix on a library package, the ServiceVersion (third part of the NuGet package version) will need to be bumped. This property is found in the library's src project. It's also possible that the property is not in that file, in which case you'll need to add it to the project and set it to 1. If the property is already in the source project, then just increment the servicing version by 1.
```

Comment:
This is still mentioning `ServicingVersion` (with a typo as `ServiceVersion`) but I see you removed that from the project.  What mechanism are we going to use for versioning.

Reply:
Given that we don't follow the versioning rules from runtime (i.e. we update the minor version, maybe even the major version in some cases), I don't think using `ServicingVersion` is appropriate here.

I imagine we could do something similar to the Azure SDK for .NET, i.e. with https://github.com/Azure/azure-sdk-for-net/commit/ad63feb27190ce523bb55adce27fd16f73a2106e.

A bot would take care of updating the package version, package validation baseline version and the assembly version (if defined in the project file). A developer only needs to set `<IsPackable>true</IsPackable>` in the project file.

While this improves the developer innerloop (single property vs multiple properties) it would not work well with incremental servicing when using live references. For that I imagine the following would work:

```
<PropertyGroup>
  <IsPackable>false</IsPackable>
  <VersionPrefix>4.6.0</VersionPrefix>
  <VersionPrefix Condition="'$(IsPackable)' == 'true'">4.7.0</VersionPrefix>
  <AssemblyVersion>4.0.6.0</AssemblyVersion>
  <AssemblyVersion Condition="'$(IsPackable)' == 'true'">4.0.7.0</AssemblyVersion>
  <PackageValidationBaselineVersion>4.6.0</PackageValidationBaselineVersion>
</PropertyGroup>
```

While that is a bit verbose (2 extra LOC), it would provide the following benefits over the existing runtime model:
1. A developer only needs to set a single property to get a library serviced.
2. By default, the patch version is incremented automatically by a bot and a developer can change that, i.e. when performing a breaking change.
3. Libraries can version independently from each other.
4. A library can version the `AssemblyVersion` and `PackageVersion` independently.

Thoughts on that?

Reply:
> A developer only needs to set a single property to get a library serviced.

Which property from your example?

Reply:
`<IsPackable>true</IsPackable>`

Reply:
> Given that we don't follow the versioning rules from runtime (i.e. we update the minor version, maybe even the major version in some cases), I don't think using ServicingVersion is appropriate here.

It can still work, it just needs a different baseline version per library.  Servicing should still only be incrementing the 3rd part of the package version, 4th part of the assembly version.  Imagine each library had committed it's InitialPackageVersion / InitalAssemblyVersion to it's props.  ServicingVersion would just increment those.  We should still avoid incrementing AssemblyVersion for libraries with shared framework overlap - we don't want these new versions to start changing the OOB behavior for frameworks where they were inbox.

I'm not sold on having a bot provides much help here - who triggers that bot and when?  Is this just a script we run to reset branding in the repo?

We should start by describing what's necessary to ship the new version.  If we want to make something the responsibility of a bot then add that on top and as a separate refinement.

Reply:
> It can still work, it just needs a different baseline version per library. Servicing should still only be incrementing the 3rd part of the package version, 4th part of the assembly version.

Any package that targets .NETCoreApp will need to update its major or minor version every other year when a TFM / runtime reaches EOL. I wouldn't want libraries to drop TFMs as part of a servicing version increment in the package version.

> I'm not sold on having a bot provides much help here - who triggers that bot and when? Is this just a script we run to reset branding in the repo?

That bot will be automated, similar to dependabot. I plan to reach out to the Azure team to get more insights into their bot. This shouldn't require any manual work.

Reply:
I'm not sure we have a great trigger in the same way that dependabot does.  You might find that such a trigger is when we want to do rebranding in the repo (which might be before we ship).  Maybe the bot becomes just a script that we run?

The ServicingVersion logic we have in dotnet/runtime could be automated by such a bot - though it does more than just making it easier to increment the version.  It also tries to hide the developer from understanding the `AssemblyVersion` rules specific to inbox / oob.  If you take that away and have the bot directly modifying `AssemblyVersion` that bot needs to know those rules.
 
I'd really like to see the end-to-end versioning process spelled out.  That should consider when to change assembly version and make sure it supports the rules we follow for .NETFramework / .NETCore -- we can't be versioning these assemblies in a way that they will start passing the versions that are inbox in later frameworks.  Many of the assemblies that make their way to this repo will have an inbox relationship so it's something we need to support.  We should file an issue for that - it doesn't need to happen as part of this PR.  

It's your call if you want to merge this with these versioning changes - to me it still seems like a sketch of a solution that replaces a different sketch 🤷